otelcol/config_test.go

```diff
@@ -59,6 +62,14 @@ func (nc *nopProcConfig) Validate() error {
 	return nc.validateErr
 }
 
+type nopConnConfig struct {
+	validateErr error
+}
+
+func (nc *nopConnConfig) Validate() error {
+	return nc.validateErr
+}
```

Comment:
Separate PR, please merge all these config structs. We merged the config some time ago, so they don't need to be a struct per component type.

Reply:
Ok, I will work on this next.